core/domain/event_services.py

```diff
@@ -162,6 +162,75 @@ def _handle_event(
             params, play_type)
 
 
+class HintRequestEventHandler(BaseEventHandler):
+    """Event handler for recording hint request events."""
+
+    EVENT_TYPE = feconf.EVENT_TYPE_HINT_REQUEST
+
+    @classmethod
+    def increment_hint_views(cls, hint_views_model, index):
+        if len(hint_views_model.viewed_hints_list) < index + 1:
+            hint_views_model.viewed_hints_list.append({
+                'num_views': 1,
+                'num_succeeds': 0
+            })
+        else:
+            hint_views_model.viewed_hints_list[index]['num_views'] += 1
+        hint_views_model.put()
+
+    @classmethod
+    def _handle_event(
+            cls, exp_id, exp_version, state_name, session_id,
+            play_type, hint_index):
+        stats_models.HintRequestEventLogEntryModel.create(
+            exp_id, exp_version, state_name, session_id,
+            play_type, hint_index)
+        cls.increment_hint_views(stats_models.HintViewsModel.get_or_create(
+            exp_id, state_name, exp_version), hint_index)
```

Comment:
In general we are doing things in the order (exp_id, exp_version, state_name), so let's keep that ordering here too. Ditto elsewhere.